### PR TITLE
Add fusion-plugin, INLINE, refactor benchmarks, use tasty-bench

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: Benchmark/System/Process.hs [hunk 1]
```diff
@@ -25,84 +21,90 @@ import qualified Streamly.Data.Fold as FL
...
+trToStderr = "./writeTrToError.sh"
+
+trToStderrContent :: String
+trToStderrContent =
+    "tr [a-z] [A-Z] <&0 >&2"
+
+createExecutable :: IO ()
+createExecutable = do
+    writeFile trToStderr trToStderrContent
+    callCommand ("chmod +x " ++ trToStderr)
```
Comment (adithyaov) [inline Benchmark/System/Process.hs:114]: Why are we creating and deleting files from the Haskell code? To keep it all in one place?

File: Benchmark/System/Process.hs [hunk 2]
```diff
@@ -16,11 +18,21 @@ import System.Process (proc, createProcess, waitForProcess, callCommand)
...
     as FH (toBytes, toChunks, putBytes, putChunks)
 
+-- XXX replace with streamly versions once they are fixed
+{-# INLINE rights #-}
+rights :: (S.IsStream t, Monad m, Functor (t m)) => t m (Either a b) -> t m b
+rights = fmap (fromRight undefined) . S.filter isRight
+
+{-# INLINE lefts #-}
+lefts :: (S.IsStream t, Monad m, Functor (t m)) => t m (Either a b) -> t m a
+lefts = fmap (fromLeft undefined) . S.filter isLeft
```
Comment (adithyaov) [inline Benchmark/System/Process.hs:34]: What's wrong with them? I've used streamly versions in tests.

**Review by adithyaov — COMMENTED** [review]
@harendra-kumar You've told me that the benchmarks were not proper. How did you fix them and what exactly was the problem?
Were we benching on small-sized streams?

File: Benchmark/System/Process.hs (hunk 1 above)
Comment (harendra-kumar) [reply Benchmark/System/Process.hs:114]: This is how it was written by the original author. I think we can create a script file and commit it.

File: Benchmark/System/Process.hs (hunk 2 above)
Comment (harendra-kumar) [reply Benchmark/System/Process.hs:34]: There is no INLINE pragma, hurting performance.

**Comment by harendra-kumar** [comment]
The streams were too small, in KBs. Also, copying from "/dev/random" did not work on Linux.

**Review by adithyaov — APPROVED** [review]

**Pull request merged**